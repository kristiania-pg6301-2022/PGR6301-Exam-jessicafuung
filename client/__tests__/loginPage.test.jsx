import {
  EndSession,
  LoginCallback,
  LoginPage,
  StartLogin,
} from "../pages/loginPage";
import React from "react";
import ReactDOM from "react-dom";
import { MemoryRouter } from "react-router-dom";

describe("login", () => {
  it("snapshot 1", () => {
    const element = document.createElement("div");
    ReactDOM.render(
      <MemoryRouter>
        <LoginPage />
      </MemoryRouter>,
      element
    );
    expect(element.innerHTML).toMatchSnapshot();
  });

  it("snapshot 2", () => {
    const element = document.createElement("div");
    ReactDOM.render(
      <MemoryRouter>
        <StartLogin />
      </MemoryRouter>,
      element
    );
    expect(element.innerHTML).toMatchSnapshot();
  });

  it("snapshot 3", () => {
    const element = document.createElement("div");
    ReactDOM.render(
      <MemoryRouter>
        <LoginCallback />
      </MemoryRouter>,
      element
    );
    expect(element.innerHTML).toMatchSnapshot();
  });

  it("snapshot 4", () => {
    const element = document.createElement("div");
    ReactDOM.render(
      <MemoryRouter>
        <EndSession />
      </MemoryRouter>,
      element
    );
    expect(element.innerHTML).toMatchSnapshot();
  });
});
