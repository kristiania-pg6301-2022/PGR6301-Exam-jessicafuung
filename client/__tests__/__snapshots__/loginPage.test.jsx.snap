// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login snapshot 1 1`] = `"<div><ul><h1>Vennligst velg innlogginsmetode: </h1><div><button>Logg inn med Google konto</button></div></ul><ul><div><button>Logg inn med ansatt-konto</button></div></ul></div>"`;

exports[`login snapshot 2 1`] = `"<div><ul><h1>Vennligst velg innlogginsmetode: </h1><div><button>Logg inn med Google konto</button></div></ul><ul><div><button>Logg inn med ansatt-konto</button></div></ul></div>"`;

exports[`login snapshot 3 1`] = `"<h1>Please wait...</h1>"`;

exports[`login snapshot 4 1`] = `"<h1>Please wait...</h1>"`;
